app/models/host/base.rb

```diff
@@ -172,6 +172,11 @@ def set_interfaces(parser)
         iface = get_interface_scope(name, attributes).first || interface_class(name).new(:managed => false)
         # create or update existing interface
         set_interface(attributes, name, iface)
+        if iface.primary?
+          self.interfaces[0] = iface
```

Comment:
I'm mostly concerned about this, while this probably works for new host, for existing one, it's unlikely that first interface must be the primary one. Sorry I don't have a working code how to do this properly atm.